src/module_4/push_exploration_non_linear.md

```diff
@@ -0,0 +1,2005 @@
+    0                            33.0                        42.0   
+    1                            33.0                        42.0   
+    2                            33.0                        42.0   
+    3                            33.0                        42.0   
+    4                            33.0                        42.0   
+    
+       std_days_to_buy_variant_id  days_since_purchase_product_type  \
+    0                   31.134053                              30.0   
+    1                   31.134053                              30.0   
+    2                   31.134053                              30.0   
+    3                   31.134053                              30.0   
+    4                   31.134053                              30.0   
+    
+       avg_days_to_buy_product_type  std_days_to_buy_product_type  
+    0                          30.0                      24.27618  
+    1                          30.0                      24.27618  
+    2                          30.0                      24.27618  
+    3                          30.0                      24.27618  
+    4                          30.0                      24.27618  
+
+
+    INFO - Length initial data: 2880549
+    INFO - Rows dropped with NA's: 0
+    INFO - Length filtered data: 2163953
+    
+    INFO - Date from: 2020-10-05
+    INFO - Date to: 2021-03-03
+    INFO - Train set ratio: 73.44%
+    INFO - Validation set ratio: 13.32%
+    INFO - Test set ratio: 13.24%
+
+
+# Non-linear models
+
+We will apply different models and compare against previous results, in the following order:
+- Decision Trees
+- Random Forests
+- Gradient Bosting Trees
+- XGBoost
+- Catboost
+
+In that sense, let's modify the generate_evaluation_curves to plot multiple funcs at the same time, in order to compare visually the performance of each model vs. baseline.
+
+
+```python
+X_train, y_train, X_val, y_val, X_test, y_test = temporal_data_split(data_processed, cols = cols)
+scaler = StandardScaler()
+X_train = scaler.fit_transform(X_train)
+X_val = scaler.fit_transform(X_val)
+```
+
+    INFO - Date from: 2020-10-05
+    INFO - Date to: 2021-03-03
+    INFO - Train set ratio: 73.44%
+    INFO - Validation set ratio: 13.32%
+    INFO - Test set ratio: 13.24%
+
+
+## Decision Tree
+
+
+```python
+import matplotlib.pyplot as plt
+from sklearn.metrics import roc_curve, auc, precision_recall_curve
+from datetime import datetime
+import os
+
+def generate_evaluation_curves_multi(
+    model_names: list, y_preds: list, y_test, save_curves_path: str = None
+):
+    """
+    Generate ROC and Precision-Recall curves for multiple binary classification models
+    and display them in a single figure.
+    """
+
+    # Create a timestamp for unique filenames
+    timestamp = datetime.now().strftime("%Y%m%d%H%M%S")
+
+    plt.figure(figsize=(10, 5))
+
+    # Plot for each model
+    for model_name, y_pred in zip(model_names, y_preds):
```

Comment:
nit pick: I suggest to keep 2 functions: the original one for the AUC for a single prediction and the one which goes through the different predictions with the for, get the metrics for each of them and joins them